hepcrawl/parsers/osti.py

```diff
@@ -0,0 +1,362 @@
+            {u'doi': value,
+             u'material': u'publication',
+             u'source': self.source}
+            for value in doi_values.split(u' ')
+        ]
+        return dois
+
+    @property
+    def osti_id(self):
+        """OSTI id for the publication.
+
+        Returns:
+            str: integer identifier as a string, e.g. u'1469753'
+        """
+        return self.record.get(u'osti_id')
+
+    @property
+    def journal_year(self):
+        """Publication year.
+
+        Returns:
+            str: 4 digit year or empty string
+        """
+        if self.date_published:
+            return self.date_published[:4]
+        return ''
+
+    @property
+    def journal_title(self):
+        """Journal name.
+
+        Returns:
+            str: name of the journal, e.g. u'Nuclear Physics. A'
+        """
+        return self.record.get(u'journal_name')
+
+    @property
+    def journal_issue(self):
+        """Journal issue.
+
+        Returns:
+            str: issue number or letter, e.g. u'34' or u'C'
+        """
+        return self.record.get(u'journal_issue')
+
+    @property
+    def journal_issn(self):
+        """Journal ISSN.
+
+        Returns:
+            str: ISSN of the journal
+        """
+        return self.record.get(u'journal_issn')
+
+    @property
+    def journal_volume(self):
+        """Journal volume.
+
+        Returns:
+           str: journal volume or volumes, e.g. u'55' or u'773-774'
+        """
+        return self.record.get(u'journal_volume')
+
+    @property
+    def language(self):
+        """Language of the publication.
+
+        Returns:
+            str: language, e.g. u'English'
+        """
+        return self.record.get(u'language', '').capitalize()
+
+    @property
+    def pageinfo(self):
+        """Format and page information.
+
+        Returns:
+            dict: lookup table for various pieces of page information
+        """
+
+        return self._get_pageinfo(self.record.get(u'format'))
+
+    @property
+    def publication_info(self):
+        """Journal publication information (pubnote).
+
+        Returns:
+            dict:
+        """
+        publication_info = {
+            u'artid': self.pageinfo.get(u'artid'),
+            u'journal_title': self.journal_title,
+            u'journal_issue': self.journal_issue,
+            u'journal_volume': self.journal_volume,
+            # 'material': self.material,
+            u'page_start': self.pageinfo.get(u'page_start'),
+            u'page_end': self.pageinfo.get(u'page_end'),
+            u'pubinfo_freetext': self.pageinfo.get(u'freeform'),
+            u'year': self.journal_year,
+        }
+        return publication_info
+
+    @property
+    def report_numbers(self):
+        """Report numbers.
+
+        Returns:
+            list: list of report number strings, e.g.
+                  [u'arXiv:1708.05642', u'FERMILAB-PUB-17-088']
+        """
+        rns = self.record.get(u'report_number')
```

Comment:
if you do
```suggestion
        rns = self.record.get(u'report_number', '')
```
you can skip the `if` afterwards.

Reply:
there is a difference, though. your code returns `['']` while my code returns `[]` 
mine is an empty list, yours is a list containing one empty string. those behave differently in e.g. `if ['']` vs `if []`